mostlyai/sdk/_data/file/base.py

```diff
@@ -365,6 +365,14 @@ def _get_dataset_format(self) -> ds.FileFormat:
     def _get_columns(self) -> list[str]:
         return [c.name for c in self.dataset.schema if c.name != ""]
 
+    def handle_if_exists(self, if_exists: str = "fail") -> str:
+        if not if_exists == "replace" and self.container.path.exists():
+            if if_exists == "fail":
+                raise MostlyDataException("Destination location already exists.")
+            if if_exists == "append" and self.IS_WRITE_APPEND_ALLOWED:
+                return "a"
+        return "w"
```

Comment:
```python
def handle_if_exists(self, if_exists: str = "fail") -> str:
    destination_exists = self.container.path.exists()
    if destination_exists:
        if if_exists == "fail":
            raise MostlyDataException("Destination location already exists.")
        elif if_exists == "append" and self.IS_WRITE_APPEND_ALLOWED:
            return "a"
    return "w"
```

Is that logic a bit more readible?

Reply:
Yes, the reason I've added `not if_exists == "replace"  and  ` is to avoid potential API calls (e.g. buckets) if we anyway don't care about its existence. But probably those are anyway marginal, given that we intend to write a file.